RFCs/0004-add-pdk-publish.md

```diff
@@ -0,0 +1,123 @@
+# Add `pdk publish` subcommand
+
+## Summary
+
+Add a new `pdk publish` subcommand that allows publishing a given module tarball to the Forge.  
+*Note that this RFC supersedes [RFC-0003](https://github.com/puppetlabs/pdk-planning/blob/master/RFCs/0003-add-pdk-release.md). While RFC-0003 may eventually still be implemented (likely as part of a PDK 2.x major release), this work represents a smaller scope that is more appropriate to the current PDK 1.x series.*
+
+## Background & Assumptions
+
+- With the addition of module management endpoints to the Forge API and authentication via API keys, authenticating and publishing to the Forge has a public and stable interface.
```

Comment:
Might be useful to add some links to the Forge API documentation for the new endpoints.

Reply:
Added links to the `POST /v3/releases` and authentication sections in the docs. There didn't seem to be a clean way of linking to all of the module management endpoints, plus it seems like maybe they're not all super relevant? Though I suppose some of the other actions could be implemented by other subcommands, which may be worth adding to the `Future considerations` section.

Reply:
Yeah, I think at least one guidepost out to the API docs is all we need.

+1 to adding the other actions to the "Future Considerations" section.